Shared/Samples/Edit features with feature-linked annotation/EditFeaturesWithFeatureLinkedAnnotationView.swift

```diff
@@ -0,0 +1,166 @@
+                .overlay(alignment: .top) {
+                    Text(instruction.message)
+                        .multilineTextAlignment(.center)
+                        .frame(maxWidth: .infinity, alignment: .center)
+                        .padding(8)
+                        .background(.regularMaterial, ignoresSafeAreaEdges: .horizontal)
+                }
+        }
+        .onChange(of: model.selectedFeature == nil) { _ in
+            if model.selectedFeature?.geometry is Point {
+                // Presents the alert to update the feature's address if the feature is a point.
+                buildingNumber = model.selectedFeatureAddress.buildingNumber
+                streetName = model.selectedFeatureAddress.streetName ?? ""
+                editAddressAlertIsPresented = true
+            } else if let polyline = model.selectedFeature?.geometry as? Polyline,
+                      polyline.parts.contains(where: { $0.points.count > 2 }) {
+                // Shows a message if the feature is a polyline with any part
+                // containing more than one segment, i.e., a curve.
+                instruction = .selectStraightPolyline
+                model.clearSelectedFeature()
+            }
+        }
+        .alert("Edit Address", isPresented: $editAddressAlertIsPresented) {
+            TextField("Building Number", value: $buildingNumber, format: .number.grouping(.never))
+                .keyboardType(.numberPad)
+            TextField("Street Name", text: $streetName)
+            Button("Cancel", role: .cancel) {
+                model.clearSelectedFeature()
+                instruction = .selectFeature
+            }
+            Button("Done") {
+                selectedAction = .setFeatureAddress(
+                    buildingNumber: buildingNumber,
+                    streetName: streetName
+                )
+            }
+        } message: {
+            Text("Edit the feature's 'AD_ADDRESS' and 'ST_STR_NAM' attributes.")
+        }
+        .alert("Confirm Move", isPresented: $moveConfirmationAlertIsPresented) {
+            Button("Cancel", role: .cancel) {
+                model.clearSelectedFeature()
+                instruction = .selectFeature
+            }
+            Button("Move") {
+                selectedAction = .updateFeatureGeometry(mapPoint: tapLocation!)
+            }
+        } message: {
+            Text("Are you sure you want to move the selected feature?")
+        }
+    }
+}
+
+private extension EditFeaturesWithFeatureLinkedAnnotationView {
+    /// An asynchronous action associated with the sample.
+    enum AsyncAction: Equatable {
+        /// Sets up the map for the sample.
+        case setUpMap
+        /// Selects a feature identified at a given point on the screen.
+        case selectFeature(screenPoint: CGPoint)
+        /// Sets the address attributes of the selected feature to given values.
+        case setFeatureAddress(buildingNumber: Int32?, streetName: String)
```

Comment:
I think this optionality can also be removed. If either the building number is `nil` or the street name is empty, we shouldn't allow the value to be edited?